forgerock-authenticator/src/main/java/org/forgerock/android/auth/AuthenticatorManager.java

```diff
@@ -128,6 +142,29 @@ boolean updateAccount(Account account) {
         }
     }
 
+    boolean lockAccount(Account account, FRAPolicy policy) throws AccountLockException {
+        if (account.isLocked()) {
+            throw new AccountLockException("Error locking the Account: Account is already locked.");
+        } else if (policy.getName() == null || policy.getName().isEmpty()) {
+            throw new AccountLockException("Error locking the Account: The policy name is required.");
+        } else if (!account.getPolicies().contains(policy.getName())) {
```

Comment:
Recommend mapping the policy name instead of the whole string.

Reply:
Not sure if I got it.

Reply:
The getPolicies() return a string of json, we check if the json contains a policy name, it may be easy to get name collision, for example, if developer create a policy called **Score**, that may collide with the jailbreakScore.

Reply:
Got it, good catch! I will fix it

Reply:
Done